contracts/adapters/StreamingFeeSplitAdapter.sol

```diff
@@ -0,0 +1,133 @@
+
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+
+import { BaseAdapter } from "../lib/BaseAdapter.sol";
+import { IBaseManager } from "../interfaces/IBaseManager.sol";
+import { ISetToken } from "../interfaces/ISetToken.sol";
+import { IStreamingFeeModule } from "../interfaces/IStreamingFeeModule.sol";
+import { PreciseUnitMath } from "../lib/PreciseUnitMath.sol";
+import { TimeLockUpgrade } from "../lib/TimeLockUpgrade.sol";
+
+
+/**
+ * @title StreamingFeeSplitAdapter
+ * @author Set Protocol
+ *
+ * Smart contract adapter that allows for splitting and setting streaming fees. Fee splits are updated by operator.
+ * Any fee updates are timelocked.
+ */
+contract StreamingFeeSplitAdapter is BaseAdapter, TimeLockUpgrade {
+    using Address for address;
+    using PreciseUnitMath for uint256;
+    using SafeMath for uint256;
+
+    /* ============ Events ============ */
+
+    event FeesAccrued(address indexed _operator, address indexed _methodologist, uint256 _operatorTake, uint256 _methodologistTake);
+    
+    /* ============ State Variables ============ */
+
+    ISetToken public setToken;
+    IStreamingFeeModule public streamingFeeModule;
+
+    // Percent of fees in precise units (10^16 = 1%) sent to operator, rest to methodologist
+    uint256 public operatorFeeSplit;
+
+    /* ============ Constructor ============ */
+
+    constructor(
+        IBaseManager _manager,
+        IStreamingFeeModule _streamingFeeModule,
+        uint256 _operatorFeeSplit
+    )
+        public
+        BaseAdapter(_manager)
+    {
+        streamingFeeModule = _streamingFeeModule;
+        operatorFeeSplit = _operatorFeeSplit;
+        setToken = manager.setToken();
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * ANYONE CALLABLE: Accrues fees from streaming fee module. Gets resulting balance after fee accrual, calculates fees for
+     * operator and methodologist, and sends to each.
+     */
+    function accrueFeesAndDistribute() public {
+        streamingFeeModule.accrueFee(setToken);
+        
+        uint256 totalFees = setToken.balanceOf(address(manager));
+        
+        address operator = manager.operator();
+        address methodologist = manager.methodologist();
+
+        uint256 operatorTake = totalFees.preciseMul(operatorFeeSplit);
+        uint256 methodologistTake = totalFees.sub(operatorTake);
+
+        if (operatorTake > 0) {
+            invokeManagerTransfer(address(setToken), operator, operatorTake);
+        }
+
+        if (methodologistTake > 0) {
+            invokeManagerTransfer(address(setToken), methodologist, methodologistTake);
+        }
+
+        emit FeesAccrued(operator, methodologist, operatorTake, methodologistTake);
+    }
+
+    /**
+     * ONLY OPERATOR: Updates streaming fee on StreamingFeeModule. NOTE: This will accrue streaming fees though not send to operator
```

Comment:
Where does it accrue to then?

Reply:
Alternative, why not have this call `accrueFeesAndDistribute` first

Reply:
Fees are auto-accrued on the `StreamingFeeModule` when the fee is updated. So they'll accrue to the manager contract. If someone wants to get them they could call `accrueFeesAndDistribute` and they can be distributed.